test/types/TestTuple.cpp

```diff
@@ -0,0 +1,41 @@
+#include "Types/Tuple.h"
+#include <Kokkos_Macros.hpp>
+#include <iostream>
+#include <memory>
+#include <Ippl.h>
+//#include <Kokkos_Core.hpp>
+#define alwaysAssert(X) do{if(!(X)){std::cerr<<"Assertion " << #X << " failed: " << __FILE__ << ": " << __LINE__ << '\n'; exit(-1);}}while(false)
```

Comment:
I would split this macro over several lines for readability.

However, this program also feels more like a unit test. Instead of building your own `assert`, you could just use the ones already in GTest.

Reply:
Ah yes, that was a bit of a hack, also I just went off tests/vector